javaslang/src/test/java/javaslang/concurrent/FutureTest.java

```diff
@@ -232,6 +231,42 @@ public void shouldCompleteWithFailureWhenExecutorServiceThrowsRejectedExecutionE
         assertFailed(future, RejectedExecutionException.class);
     }
 
+    @Test
+    public void shouldWaitGivenDelayBeforeComputingUsingDefaultExecutorService() {
+        final long delay = 250;
+        final long start = System.currentTimeMillis();
+        final Future<Long> future = Future.of(() -> System.currentTimeMillis() - start, delay);
+        waitUntil(future::isCompleted);
+        assertThat(future.get()).isGreaterThanOrEqualTo(delay);
+    }
+
+    @Test
+    public void shouldWaitGivenDelayBeforeComputingUsingSpecificExecutorService() {
+        final long delay = 250;
+        final long start = System.currentTimeMillis();
+        final Future<Long> future = Future.of(waitingExecutorService(100), () -> System.currentTimeMillis() - start, delay);
+        waitUntil(future::isCompleted);
+        assertThat(future.get()).isGreaterThanOrEqualTo(delay);
+    }
+
+    @Test
+    public void shouldWaitGivenDelayBeforeRunningUsingDefaultExecutorService() {
+        final long delay = 250;
+        final long[] time = new long[] { System.currentTimeMillis() };
+        final Future<Void> future = Future.run(() -> time[0] = System.currentTimeMillis() - time[0], delay);
+        waitUntil(future::isCompleted);
+        assertThat(time[0]).isGreaterThanOrEqualTo(delay);
+    }
+
+    @Test
+    public void shouldWaitGivenDelayBeforeRunningUsingSpecificExecutorService() {
+        final long delay = 250;
+        final long[] time = new long[] { System.currentTimeMillis() };
```

Comment:
you could use the built-in array initializer here (and in other places in the test):
```java
final long[] time = { System.currentTimeMillis() };
```